### PR TITLE
Fix BatchNorm for model cloning

### PR DESCRIPTION
**Pull request by yanndupis**
Hey @mortendahl,

It's a WIP PR. This bug #705 is a bit more complicated than I expected. 

When we clone the a model, first we instantiate the model. During this process, the weights (TFE variables) from each layer gets stored in a `self.weights` model attribute. Then with `set_weights` we assign the numpy weights to the TFE variables.

The difference with Batchnorm is that the weights are not private variables but public variables (e.g moving_variance) because it involves sqrt, division etc. 

The problem is that we are less equipped to assign numpy weights (define_public_placeholder) to `PublicPondVariable`.

### CODE REVIEW

File: tf_encrypted/keras/layers/normalization.py [hunk 1]
```diff
@@ -161,22 +165,26 @@ def build(self, input_shape):
...
+    self.moving_mean = self.add_weight(moving_mean, make_private=False)
 
     moving_variance = self.moving_variance_initializer(param_shape)
+    moving_variance = self.add_weight(moving_variance, make_private=False)
 
-    denomtemp = 1.0 / tf.sqrt(moving_variance + self.epsilon)
+    # to_navive() tranformation is bad and not yet working
+    # Find solution to compute tf.sqrt on PondPublicVariable
+    # or use different approach
+    denomtemp = 1.0 / tf.sqrt(moving_variance.to_native(), + self.epsilon)
```
Comment (yanndupis) [inline tf_encrypted/keras/layers/normalization.py]: really need to find a solution to this stuff. the challenge is that Keras weights is moving_variance when our public weight is `denomtemp` because of division/sqrt

**Comment by yanndupis** [comment]
Hey @mortendahl, I think it's ready for review :). Thank you

**Comment by yanndupis** [comment]
Thanks for the review! Sounds good, let's wait to settle on this.

**Comment by yanndupis** [comment]
Most of the comments should be addressed. Thanks
